### PR TITLE
StorageCnchMergeTree properly supports skipping indices

### LINKED ISSUE #102: Several issues with secondary indexes
(you don't have to strictly follow this form)

## Bug Report
1. Secondary index is not used in query

```sql
create table t (x UInt32, y UInt32, index minmax_y y type minmax granularity 1) engine=CnchMergeTree order by x;
insert into t select (number ,number) from numbers(100000);
select count() from t where y = 1;
```
The select query should read only 1 mark, currently doing full-scan:

image

Log from worker:

image

2. Materialize index has't supported

image

### PR DESCRIPTION
**Pull request by canhld94**
I hereby agree to the terms of the CLA available at: [placeholder]

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

- StorageCnchMergeTree supports using skipping indices in query
- StorageCnchMergeTree supports `ALTER table MATERIALIZE INDEX ...`

Closes #102 

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ByConity Documentation](https://github.com/ByConity/ByConity/blob/master/CONTRIBUTING.md) guide first.

### CODE REVIEW

**Comment by canhld94** [comment]
@Clark0 I temporary disable adaptive index granularity tests, seem it's not working with secondary indexes

**Comment by canhld94** [comment]
@hustnn @Clark0 @linkwk7 pls help to review, thanks!

File: tests/queries/4_cnch_stateless/00974_adaptive_granularity_secondary_index.sql [hunk 1]
```diff
@@ -1,10 +1,9 @@
-
 DROP TABLE IF EXISTS indexed_table;
 CREATE TABLE indexed_table
 (    `tm` DateTime,
     `log_message` String,
     INDEX log_message log_message TYPE tokenbf_v1(4096, 2, 0) GRANULARITY 1
-)ENGINE = CnchMergeTree ORDER BY (tm) SETTINGS index_granularity_bytes = 50; --, min_index_granularity_bytes = 40;
+) ENGINE = CnchMergeTree ORDER BY (tm); -- SETTINGS index_granularity_bytes = 50; --, min_index_granularity_bytes = 40;
```
Comment (Clark0) [inline tests/queries/4_cnch_stateless/00974_adaptive_granularity_secondary_index.sql:6]: Does the table with adaptive index granularity support secondary index now?

Comment (canhld94) [reply tests/queries/4_cnch_stateless/00974_adaptive_granularity_secondary_index.sql:6]: No, loading marks of index files has an exception if enable adaptive index granularity

**Review by Clark0 — APPROVED** [review]

**Pull request merged**